From 279a797fb765056867a115846ccf1548747457e4 Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Thu, 8 Nov 2018 11:23:19 +0200
Subject: [PATCH 0207/1239] arm64: mvebu: a8k: Simplify the SoC information
 index lookup

Simplify the logic of get_soc_table_index() function code
for easier extension in the future.

Change-Id: Iee2f3ab5cb43073d80402fe781a1d963b2ac986f
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
---
 arch/arm/mach-mvebu/armada8k/soc.c | 29 ++++++++++++-----------------
 1 file changed, 12 insertions(+), 17 deletions(-)

diff --git a/arch/arm/mach-mvebu/armada8k/soc.c b/arch/arm/mach-mvebu/armada8k/soc.c
index f4b6913690..223e5873bb 100644
--- a/arch/arm/mach-mvebu/armada8k/soc.c
+++ b/arch/arm/mach-mvebu/armada8k/soc.c
@@ -103,23 +103,18 @@ static int get_soc_table_index(u32 *index)
 	get_soc_type_rev(&soc_type, &rev);
 	get_ap_soc_type(&ap_type);
 
-	for (i = 0; i < sizeof(soc_info_table) / sizeof(struct soc_info); i++) {
-		if ((soc_type ==
-			soc_info_table[i].soc.module_type) &&
-		   (rev == soc_info_table[i].soc.module_rev) &&
-		    ap_type == soc_info_table[i].ap.module_type) {
-			if (!get_soc_sub_rev(&sub_rev)) {
-				if (sub_rev == soc_info_table[i].sub_rev) {
-					*index = i;
-					ret = 0;
-				} else {
-					continue;
-				}
-			} else {
-				*index = i;
-				ret = 0;
-			}
-		}
+	for (i = 0; i < ARRAY_SIZE(soc_info_table) && ret != 0; i++) {
+		if ((soc_type != soc_info_table[i].soc.module_type) ||
+		    (rev != soc_info_table[i].soc.module_rev) ||
+		    ap_type != soc_info_table[i].ap.module_type)
+			continue;
+
+		if (!get_soc_sub_rev(&sub_rev) &&
+		    (sub_rev != soc_info_table[i].sub_rev))
+			continue;
+
+		*index = i;
+		ret = 0;
 	}
 
 	if (ret)
-- 
2.29.0

